src/test/ui/rfc-2091-track-caller/warn-fn-prototype.rs

```diff
@@ -0,0 +1,16 @@
+// run-pass
+
+#![feature(track_caller)]
+#![allow(dead_code)]
+
+trait Trait {
+    #[track_caller] //~ WARN: `#[track_caller]` is ignored on function prototypes
+    fn foo(&self);
```

Comment:
We might also want to error here if we want to reserve the right to add meaning to it?

Reply:
It does seem prudent to error for now, but I'm curious, do you have thoughts re. meaning?

Reply:
It could be inherited by `impl`s by default, for example.

Reply:
That seems plausible, but on the other hand, what is the remedy users would have to disable that effect on specific impls then?

Reply:
They can introduce a non-`#[track_caller]` wrapper, the same way e.g. `libcore` introduces `#[cold]` `#[inline(never)]` functions around panics (although those serve a different purpose).

Reply:
I'd be very interested in a version of track_caller that's automatically applied to all implementors of a method -- happy to promote warning to error if it creates room for that down the road.

Reply:
@anp If you want to follow up with such a PR we could also deal with it before stabilization if you like so that everything is more "complete" when we do stabilize.

Reply:
Cool. It occurs to me from a recent conversation at work that a blanket track_caller in traits would (I think?) allow error traits to automatically record callchain information with `?`. Exciting if true.